gossipd/gossipd_wire.csv

```diff
@@ -15,6 +15,11 @@ msgdata,gossipd_init,announcable,wireaddr,num_announcable
 msgdata,gossipd_init,dev_gossip_time,?u32,
 msgdata,gossipd_init,dev_fast_gossip,bool,
 msgdata,gossipd_init,dev_fast_gossip_prune,bool,
+msgdata,gossipd_init,has_offer,bool,
+msgdata,gossipd_init,lease_proportional_basis,u16,
+msgdata,gossipd_init,lease_base_sat,amount_sat,
+msgdata,gossipd_init,channel_proportional_basis,u16,
+msgdata,gossipd_init,channel_base_msat,amount_msat,
```

Comment:
Yech.  The alternate approach is to implement setliquidyoffer RPC, and plumb through to gossipd.

Then change gossipd:
1. change update_own_node_announcement() to take a "bool startup" flag.
2. change nannounce_different to take a `bool *only_missing_liquidityoffer`, which it sets if it returns true but the only difference is that the proposed nannounce is missing a liquidity offer.
3. If "startup" and the only difference is that we don't have a liquidity offer yet, set a timer to call update_own_node_announcement(startup=false) in 60 seconds (in case they restart without the liquidity plugin, we'll then send out a new node ann).
3. Call update_own_node_announcement(startup=false) when receiving a new liquidity ann.

This seems much cleaner (ok, except for the startup delay).

Reply:
Right, I decided against that because I didn't want to expose an RPC to set the liquidityoffer on lightningd, as that's misleading. Your node physically *can't* fulfill any liquidity offer that you set on lightningd.

This design isn't perfect but it does have the advantage that the *only* way to set/update a liquidity offer is via a plugin that supports fulfilling them. This heads off any potential confusion that might come from looking at the RPCs lightningd supports and thinking you can call `setliquidityoffer` and get the ball rolling...